library/src/main/java/com/schibsted/spain/barista/cleardata/ClearFilesRule.java

```diff
@@ -0,0 +1,33 @@
+package com.schibsted.spain.barista.cleardata;
+
+import android.content.Context;
+import android.support.test.InstrumentationRegistry;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+/**
+ * This rule clears all app's stored files before running each test
+ */
+public class ClearFilesRule implements TestRule {
+
+  @Override
+  public Statement apply(final Statement base, Description description) {
+    return new Statement() {
+      @Override
+      public void evaluate() throws Throwable {
+        clearData();
+        base.evaluate();
+        clearData();
+      }
+    };
+  }
+
+  private void clearData() {
+    FileOperations.deleteAllFiles(getAppContext());
```

Comment:
Uhm.

Context has access to many dirs:

![captura de pantalla 2017-06-01 a las 16 55 51](https://cloud.githubusercontent.com/assets/887462/26685736/3bb7db2a-46eb-11e7-97e5-73b8f4312c20.png)

What do you think allowing dir to be pass through constructor? So it will look like:

```java
File FILES_DIR = InstrumentationRegistry.getTargetContext().getFilesDir();
@Rule public ClearFilesRule clearFilesRule = new ClearFilesRule(FILES_DIR);
```

For example: i use `Context.getExternalCacheDir()` many times, and it could be great to delete files there too